src/main/java/com/github/gkttk/sixth/bookdao/dao/specification/sort/AuthorSortSpecification.java

```diff
@@ -0,0 +1,18 @@
+package com.github.gkttk.sixth.bookdao.dao.specification.sort;
+
+import com.github.gkttk.sixth.bookdao.model.Book;
+import com.github.gkttk.sixth.bookdao.logic.comparator.AuthorBookComparator;
+
+import java.util.Comparator;
+
+/**
+ * Sort by author.
+ */
+public class AuthorSortSpecification extends AbstractSortSpecification {
```

Comment:
in fact you don't need SortSpecification hierarchy, just create proper comparator in dao using factory